client/tx/tx.go

```diff
@@ -246,14 +260,28 @@ func Sign(txf Factory, name string, txBuilder client.TxBuilder, overwriteSig boo
 		Data:     &sigData,
 		Sequence: txf.Sequence(),
 	}
+
 	var prevSignatures []signing.SignatureV2
 	if !overwriteSig {
 		prevSignatures, err = txBuilder.GetTx().GetSignaturesV2()
 		if err != nil {
 			return err
 		}
 	}
-	if err := txBuilder.SetSignatures(sig); err != nil {
+
+	// Overwrite or append signer infos.
+	var sigs []signing.SignatureV2
+	if overwriteSig {
+		sigs = []signing.SignatureV2{sig}
+	} else {
+		sigs = append(prevSignatures, sig)
+	}
+
+	if err := txBuilder.SetSignatures(sigs...); err != nil {
+		return err
+	}
+
+	if err := checkMultipleSigners(signMode, txBuilder.GetTx(), len(prevSignatures)); err != nil {
```

Comment:
I believe `len(prevSignatures)` is wrong. We have the signerIndex at L216, no?

Reply:
Are you saying about this? 
https://github.com/cosmos/cosmos-sdk/blob/9e6481aeb6c1986ff1406e4a230d23f68feaad13/client/tx/tx.go#L230

I checked with this, but if the current pubkey didn't exist in the `pubkeys` then the singerIndex will be always `0` (assume we are signing twice) hence it is failing in some tests.

in this case the following condition never met:
`signerIndex == (len(tx.GetSigners()) - 1)`

Reply:
Let's handle this in https://github.com/cosmos/cosmos-sdk/issues/10691